lib/v2/science/blogs.js

```diff
@@ -0,0 +1,55 @@
+const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+const { baseUrl } = require('./utils');
+
+module.exports = async (ctx) => {
+    const { name = 'pipeline' } = ctx.params;
+    const link = `${baseUrl}/blogs/${name}/feed`;
+
+    const response = await ctx.cache.tryGet(link, async () => {
+        const browser = await require('@/utils/puppeteer')();
+        const page = await browser.newPage();
+        await page.setRequestInterception(true);
+
+        page.on('request', (request) => {
+            request.resourceType() === 'document' ? request.continue() : request.abort();
+        });
+
+        await page.goto(link, {
+            waitUntil: 'domcontentloaded',
+        });
+
+        const response = await page.content();
+
+        page.close();
+        browser.close();
+        return response;
+    });
+
+    const $ = cheerio.load(response, { xmlMode: true });
+    const items = $('item')
+        .toArray()
+        .map((item) => {
+            item = $(item);
+            return {
+                title: item.find('title').text().trim(),
+                link: item.find('link').text().trim(),
+                author: item.find('dc\\:creator').text().trim(),
+                pubDate: parseDate(item.find('pubDate').text().trim()),
+                description: item.find('content\\:encoded').text().trim(),
+            };
+        });
+
+    // The RSS feed is implemented by a keyword search on the science.org end
+    // so the description field of the feed looks like this:
+    const name_re = /Keyword search result for Blog Series: (?<blog_name>[^-]+) --/;
+    const { blog_name = 'Unknown Title' } = $('channel > description').text().match(name_re).groups;
+
+    ctx.state.data = {
+        title: `Science Blogs: ${blog_name}`,
+        image: `${baseUrl}/apple-touch-icon.png`,
```

Comment:
```suggestion
        description: $('channel > description').text().replace('Keyword search result for Blog Series:', ''),
        image: `${baseUrl}/apple-touch-icon.png`,
```

Reply:
The value of `$('channel > description').text()` can be either "Keyword search result for Blog Series: In the Pipeline -- New results matching your keyword search query (updated daily)." or "Keyword search result for Blog Series: Editor's Blog -- New results matching your keyword search query (updated daily)." so  we need to remove a bit from both ends. 

If you don't like the regex how about this:
```suggestion
        description: $('channel > description').text().slice(39, -67),
        image: `${baseUrl}/apple-touch-icon.png`,
```

Reply:
Oh sorry I misunderstood, $('channel > description').text() just contains the title of the blog, sadly there is no description information in the rss feed.